tests/python/test_build.py

```diff
@@ -31,11 +34,293 @@ def test_build_cpp() -> None:
+
+    # Test square: int -> int
+    assert mod.square(5) == 25
+    metadata = mod.get_function_metadata("square")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[int], int]"
+
+    # Test reciprocal: float -> float
+    result = mod.reciprocal(2.0)
+    assert abs(result - 0.5) < 0.001
+    metadata = mod.get_function_metadata("reciprocal")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[float], float]"
+
+    # Test negate: bool -> bool
+    assert mod.negate(True) is False
+    assert mod.negate(False) is True
+    metadata = mod.get_function_metadata("negate")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[bool], bool]"
+
+    # Test uppercase_first: String -> String
+    result = mod.uppercase_first("hello")
+    assert result == "Hello"
+    metadata = mod.get_function_metadata("uppercase_first")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[str], str]"
+
+    # Test weighted_sum: int, float -> float
+    result = mod.weighted_sum(10, 2.5)
+    assert abs(result - 25.0) < 0.001
+    metadata = mod.get_function_metadata("weighted_sum")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[int, float], float]"
+
+    # Test repeat_string: String, int -> String
+    result = mod.repeat_string("ab", 3)
+    assert result == "ababab"
+    metadata = mod.get_function_metadata("repeat_string")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[str, int], str]"
+
+    # Test format_data: bool, int, float, String -> String
+    result = mod.format_data(True, 42, 3.14, "test")
+    assert "test:" in result
+    assert "flag=true" in result
+    assert "count=42" in result
+    assert "value=3.14" in result
+    metadata = mod.get_function_metadata("format_data")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[bool, int, float, str], str]"
+
+    # Test double_tensor: Tensor, Tensor -> None
+    x = numpy.array([1.0, 2.0, 3.0], dtype=numpy.float32)
+    y = numpy.empty_like(x)
+    mod.double_tensor(x, y)
+    numpy.testing.assert_allclose(y, x * 2.0)
+    metadata = mod.get_function_metadata("double_tensor")
+    assert metadata is not None
+    schema = TypeSchema.from_json_str(metadata["type_schema"])
+    assert str(schema) == "Callable[[Tensor, Tensor], None]"
+
+    # Explicitly cleanup all objects before module unload to avoid use-after-free
+    del metadata, schema, result, x, y, mod
+    gc.collect()
```

Comment:
I'm thinking if `gc` is not actually useful as TVM-FFI will cache the loaded DSOs IIRC

Reply:
Likely I’m wrong